test/ConstructiveSolidGeometry/CSG_test.jl

```diff
@@ -22,13 +25,123 @@ using SolidStateDetectors.ConstructiveSolidGeometry:
                 @test norm(cp1) == sqrt(T(14))
             end
 
+            @testset "Angular Intervals" begin
+                @testset "FullCircle" begin
+                    full_2π = T(0)..T(2π)
+                    @test _in_angular_interval_closed(T(0), full_2π)
+                    @test _in_angular_interval_closed(T(2π), full_2π)
+                    @test _in_angular_interval_closed(T(4π), full_2π)
+                    @test _in_angular_interval_closed(T(-π), full_2π)
+                    @test _in_angular_interval_closed(T(-2π), full_2π)
+                    @test !_in_angular_interval_open(T(0), full_2π)
+                    @test !_in_angular_interval_open(T(2π), full_2π)
+                    @test !_in_angular_interval_open(T(4π), full_2π)
+                    @test _in_angular_interval_open(T(-π), full_2π)
+                    @test !_in_angular_interval_open(T(-2π), full_2π)
+                end
+                @testset "Partial Circle" begin
+                    partial_2π = T(0)..T(π)
+                    @test _in_angular_interval_closed(T(0), partial_2π)
+                    @test _in_angular_interval_closed(T(2π), partial_2π)
+                    @test _in_angular_interval_closed(T(π), partial_2π)
+                    @test _in_angular_interval_closed(T(3π), partial_2π)
+                    @test _in_angular_interval_closed(T(-π), partial_2π)
+                    @test !_in_angular_interval_closed(T(-π/2), partial_2π)
+                    @test _in_angular_interval_closed(T(π/2), partial_2π)
+                    @test _in_angular_interval_closed(T(-3π/2), partial_2π)
+                    @test !_in_angular_interval_open(T(0), partial_2π)
+                    @test !_in_angular_interval_open(T(π), partial_2π)
+                    #@test !_in_angular_interval_open(T(3π), partial_2π)
+                    @test !_in_angular_interval_open(T(-π), partial_2π)
+                    @test _in_angular_interval_open(T(π/2), partial_2π)
+                    @test _in_angular_interval_open(T(-3π/2), partial_2π)
+                end
+                @testset "Negative Interval" begin
+                    neg_interval = -T(π/2)..T(π/2)
+                    @test _in_angular_interval_closed(T(0), neg_interval)
+                    @test _in_angular_interval_closed(T(2π), neg_interval)
+                    @test !_in_angular_interval_closed(T(π), neg_interval)
+                    @test !_in_angular_interval_closed(T(3π), neg_interval)
+                    @test !_in_angular_interval_closed(T(-π), neg_interval)
+                    @test _in_angular_interval_closed(T(-π/2), neg_interval)
+                    #@test _in_angular_interval_closed(T(-3π/2), neg_interval)
```

Comment:
I guess these are the cases where it fails for `Float32`?

Reply:
Exactly!

Reply:
But these are all values smaller than `0` or bigger than `2π`. I would say we can live with this for now and have a look at these cases once they might become crucial!